src/App.tsx

```diff
@@ -1,26 +1,314 @@
-/* eslint-disable max-len */
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  getTodos,
+  USER_ID,
+  deleteTodo,
+  addTodo,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Error } from './types/enumError';
+import { FilterName } from './types/enumFilterName';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMes } from './components/ErrorMes';
 
-const USER_ID = 0;
+export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Error>(Error.Default);
+  const [filteredBy, setFilteredBy] = useState<FilterName>(FilterName.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const [editingTitle, setEditingTitle] = useState<string>('');
```

Comment:
if u use (string, number, boolean) as the initial value - u don't need a generic type
```suggestion
  const [editingTitle, setEditingTitle] = useState('');
```